pkg/resource/feature_group/manager_test_suite_test.go

```diff
@@ -0,0 +1,167 @@
+	}
+	//TODO: use reflection, template to auto generate this block/method.
+	switch apiName {
+	case "CreateFeatureGroupWithContext":
+		var output svcsdk.CreateFeatureGroupOutput
+		return &output, nil
+	case "DeleteFeatureGroupWithContext":
+		var output svcsdk.DeleteFeatureGroupOutput
+		return &output, nil
+	case "DescribeFeatureGroupWithContext":
+		var output svcsdk.DescribeFeatureGroupOutput
+		return &output, nil
+	}
+	return nil, errors.New(fmt.Sprintf("no matching API name found for: %s", apiName))
+}
+
+func (d *testRunnerDelegate) Equal(a acktypes.AWSResource, b acktypes.AWSResource) bool {
+	ac := a.(*resource)
+	bc := b.(*resource)
+	// Ignore LastTransitionTime since it gets updated each run.
+	opts := []cmp.Option{cmpopts.EquateEmpty(), cmpopts.IgnoreFields(ackv1alpha1.Condition{}, "LastTransitionTime")}
+
+	if cmp.Equal(ac.ko.Status, bc.ko.Status, opts...) {
+		return true
+	} else {
+		fmt.Printf("Difference (-expected +actual):\n\n")
+		fmt.Println(cmp.Diff(ac.ko.Status, bc.ko.Status, opts...))
+		return false
+	}
+}
+
+// Checks to see if the given yaml file, with name stored as expectation,
+// matches the yaml marshal of the AWSResource stored as actual.
+func (d *testRunnerDelegate) YamlEqual(expectation string, actual acktypes.AWSResource) bool {
+     	// Get the file name of the expected yaml.
+	expectedYamlFileName := TestDataDirectory + "/" + expectation
+
+	// Build a tmp file for the actual yaml.
+	actualResource := actual.(*resource)
+	actualYamlByteArray, _ := yaml.Marshal(actualResource.ko)
+	actualYamlFileName := buildTmpFile("actualYaml", actualYamlByteArray)
+	defer os.Remove(actualYamlFileName)
+	if "" == actualYamlFileName {
+	  fmt.Printf("Could not create temporary actual file.\n")
+	  return false
+	}
+
+	// Replace Timestamps that would show up as different.
+	_, err := exec.Command("sed", "-r", "-i", ReplaceTimestampRegExp, actualYamlFileName).Output()
+	if !checkExecCommandError(err) {
+	    return false
+	}
+	
+	output,err := exec.Command("diff", "-c", expectedYamlFileName, actualYamlFileName).Output()
+	if !checkExecCommandError(err) {
+	   return false
+	}
+
+	if len(output) > 0 {
+	   actualOutput,_ := exec.Command("cat", actualYamlFileName).Output()
+	   fmt.Printf("\nExpected Yaml File Name: " + expectedYamlFileName + "\n")
+	   fmt.Printf("\nActual Output Yaml:\n" + string(actualOutput) + "\n")
+	   fmt.Printf("Diff From Expected:\n" + string(output) + "\n")
+	   return false
+	}
+	return true
+}
+
+func buildTmpFile(fileNameBase string, contents []byte) string {
+     newTmpFile, err := ioutil.TempFile(TestDataDirectory, fileNameBase)
+     if err != nil {
+        fmt.Println(err)
+	return ""
+     }
+     if _, err := newTmpFile.Write(contents); err != nil {
+        fmt.Println(err)
+	return ""
+     }
+     if err := newTmpFile.Close(); err != nil {
+        fmt.Println(err)
+	return ""
+     }		
+     return newTmpFile.Name()
+}
+
+func checkExecCommandError(err error) bool {
```

Comment:
nit: `isExecCmdError` might be more readable and return true when there is an actual error instead of false